src/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/Models/PropertyTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Linq;
+using NUnit.Framework;
+using Umbraco.Core.Models;
+using Umbraco.Tests.Common.Builders;
+using Umbraco.Tests.Common.Builders.Extensions;
+
+namespace Umbraco.Tests.UnitTests.Umbraco.Infrastructure.Models
+{ 
+    [TestFixture]
+    public class PropertyTests
+    {
+        private readonly PropertyBuilder _builder = new PropertyBuilder();
+
+        [Test]
+        public void Can_Deep_Clone()
+        {
+            var property = BuildProperty();
+
+            property.SetValue("hello");
+            property.PublishValues();
+
+            var clone = (Property)property.DeepClone();
+
+            Assert.AreNotSame(clone, property);
+            Assert.AreNotSame(clone.Values, property.Values);
+            Assert.AreNotSame(property.PropertyType, clone.PropertyType);
+            for (var i = 0; i < property.Values.Count; i++)
+            {
+                Assert.AreNotSame(property.Values.ElementAt(i), clone.Values.ElementAt(i));
+            }
+
+            // This double verifies by reflection
+            var allProps = clone.GetType().GetProperties();
+            foreach (var propertyInfo in allProps)
+            {
+                Assert.AreEqual(propertyInfo.GetValue(clone, null), propertyInfo.GetValue(property, null));
+            }
+        }
+
+        private Property BuildProperty()
+        {
+            return _builder
+                .WithId(4)
+                .WithKey(Guid.NewGuid())
+                .WithCreateDate(DateTime.Now)
```

Comment:
We can use defalts for some of these.. `Key`, Dates, ...

Reply:
Agreed. have removed.